src/kOS/Function/Misc.cs

```diff
@@ -322,7 +322,9 @@ public override void Execute(SharedObjects shared)
                 // or to a file to save:
                 if (justCompiling)
                 {
-                    List<CodePart> compileParts = shared.ScriptHandler.Compile(path, 1, fileContent.String, String.Empty, options);
+                    // since we've already read the file content, use the volume from outPath instead of the source path
+                    volume = shared.VolumeMgr.GetVolumeFromPath(outPath);
+                    List<CodePart> compileParts = shared.ScriptHandler.Compile(path, 1, fileContent.String, string.Empty, options);
                     VolumeFile written = volume.SaveFile(outPath, new FileContent(compileParts));
```

Comment:
Hmm.  The need to do this makes me think we should have a more universal API that really does work properly with a global path.  I mean, the path *contains* "0:" or "1:" in it, so our API method should know which volume it's meant to be saved onto, rather than having to be told which volume it has to be saved to.